lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -131,11 +132,18 @@ def capture(amount, authorization, options={})
       end
 
       def refund(amount, authorization, options={})
-        if auth_was_for_cim?(authorization)
+        response = if auth_was_for_cim?(authorization)
           cim_refund(amount, authorization, options)
         else
           normal_refund(amount, authorization, options)
         end
+
+        return response if response.success?
+        return response unless options[:force_full_refund_if_unsettled]
+
+        if response.params["response_reason_code"] == INELIGIBLE_FOR_ISSUING_CREDIT_ERROR
```

Comment:
Yep, because there's an `expects(:ssl_post)` 
If void (or anything) was called after the refund response, the test would 💥